### PR TITLE
Feature/151168142/view game log

### PR DESCRIPTION
**Pull request by dinobi**
#### What does this PR do?
Allows a registered user view a "game log" of previously played games.
#### Description of Task to be completed?
When a user signup or signin, the user should be able to view a "game log" of previously played games if any, and reference the friends they played with, number of rounds and overall winner of the different games played.
#### How should this be manually tested?
On the dashboard click on "view-log" button
#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
Story Title: Registered users should be able to view a a "game log" of past games
Story ID: #151168142
Story Type: Feature
#### Screenshots (if appropriate)
localhost-3000-

#### Questions:

### CODE REVIEW

File: app/models/game.js [hunk 1]
```diff
@@ -3,18 +3,17 @@
 */
 const mongoose = require('mongoose');
 
-const { Schema } = mongoose;
```
Comment (segunolalive) [inline app/models/game.js]: Why did you replace the es6 syntax with es5

**Review by segunolalive — APPROVED** [review]

File: app/models/game.js (hunk 1 above)
Comment (dinobi) [reply app/models/game.js]: I was troubleshooting then.

File: app/controllers/game.js [hunk 2]
```diff
@@ -1,20 +1,62 @@
...
+      game.userId = req.user.userId;
+      game.gameId = gameId;
+      game.save((err) => {
+        if (err) {
+          return res.status(500).json({
+            message: 'An error occured while trying to save',
+            error: err
+          });
+        }
+        return res.status(200).json(game);
```
Comment (nebanat) [inline app/controllers/game.js]: change the status code to 201

File: config/middlewares/authUser.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    return res.status(401).json('Unauthorized: No access token provided');
+  }
+  jwt.verify(token, process.env.JWT_SECRET, (error, decoded) => {
+    if (error) {
+      return res.status(403).json('Invalid token');
+    }
+    User.findById(decoded.userId)
+      .then((user) => {
+        if (!user) {
+          return res.status(404)('User not found');
```
Comment (nebanat) [inline config/middlewares/authUser.js]: This will error it should be res.status(404).send({message:'User not found'})

**Review by nebanat — CHANGES_REQUESTED** [review]

File: app/models/game.js (hunk 1 above)
Comment (segunolalive) [reply app/models/game.js]: okay. Do fix it

File: config/middlewares/authUser.js (hunk 3 above)
Comment (dinobi) [reply config/middlewares/authUser.js]: got you

File: app/controllers/game.js (hunk 2 above)
Comment (dinobi) [reply app/controllers/game.js]: ok

File: config/middlewares/authUser.js (hunk 3 above)
Comment (dinobi) [reply config/middlewares/authUser.js]: done

File: app/controllers/game.js (hunk 2 above)
Comment (dinobi) [reply app/controllers/game.js]: done

File: app/models/game.js (hunk 1 above)
Comment (dinobi) [reply app/models/game.js]: done

**Review by nebanat — APPROVED** [review]

**Review by ibonly — APPROVED** [review]

**Pull request merged**